cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java

```diff
@@ -125,14 +109,31 @@ private String getAgentArgs() {
     }
 
     private Path getTargetProfilerPath(Project project) {
-        return getProjectPath(project).resolve("Profiler" + BLANG_COMPILED_JAR_EXT);
+        return getProfilerPath(project).resolve("Profiler" + BLANG_COMPILED_JAR_EXT);
+    }
+
+    private Path getProfilerPath(Project project) {
+        try {
+            Target target = new Target(getTargetPath(project));
+            return target.getProfilerPath();
+        } catch (IOException e) {
+            throw createLauncherException("error while creating profiler directory: ", e);
+        }
     }
 
-    private Path getProjectPath(Project project) {
-        // If the --output flag is not set, create the executable in the current directory
+    private Path getTargetPath(Project project) {
         if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
             return Paths.get(System.getProperty(USER_DIR));
         }
-        return project.targetDir().resolve("bin");
+        return project.targetDir();
+    }
+
+    private String getTargetFilePath(Project project) {
+        if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
+            return Paths.get(System.getProperty(USER_DIR)).resolve(
+                    getFileNameWithoutExtension(project.sourceRoot()) + BLANG_COMPILED_JAR_EXT).toUri().getPath();
```

Comment:
Shall we check why this line is not covered?

Reply:
When running the added single file test locally, those lines are hit.